include/avif/avif.h

```diff
@@ -625,6 +629,30 @@ typedef struct avifGainMapMetadataDouble
 // to the closest uint32_t fractions.
 // Returns AVIF_FALSE if some field values are < 0 or > UINT32_MAX.
 AVIF_API avifBool avifGainMapMetadataDoubleToFractions(avifGainMapMetadata * dst, const avifGainMapMetadataDouble * src);
+
+// Performs tone mapping on a base image using the provided gain map.
```

Comment:
I am wondering if tone mapping is within the scope of libavif. Should we first add these functions to aviftest_helpers.* if we will only be using these functions in tests?

Reply:
Fair enough. I moved it to internal.h for now.
It's a lot easier to move things between avif.h and internal.h than to aviftest_helpers.

Reply:
I was chatting with Frank about this and he pointed out that tone mapping would be useful for things like converting an SDR+gainmap image to HDR+gainmap. It would not necessarily need to be in the API but at least in the tools? (to be discussed)
I will prepare a short doc so we can discuss what should and should not be in scope for libavif/avifenc/avifdec.

Reply:
I'm sorry that my comment was ambiguous. I meant to say "I am wondering if tone mapping is within the scope of the libavif **library**."

Re: declaring these two functions in internal.h: The code in the apps directory cannot use the internal functions of the libavif library. So if these two functions will be used by avifdec or avifenc, they need to be either public functions of the libavif library or functions in the apps/shared directory (i.e., the apps_shared library).

If we anticipate that these two functions will be useful to most libavif clients (such as Chrome), then they should be public functions of the libavif library. Otherwise it is better to add them to the apps/shared directory.

Reply:
Would it make sense to try and get this submitted as is in gainmap.c/internal.h and move it later once we figure out where it should be?

Reply:
Yes. I may not be able to review this pull request quickly, because I was just checking changes to the public header (avif.h). I can try to review this later today.